huaweicloud/resource_huaweicloud_rds_instance_v3.go

```diff
@@ -255,106 +232,259 @@ func resourceRdsInstanceV3() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
-			"time_zone": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				Computed: true,
-			},
 		},
 	}
 }
 
-func resourceRdsInstanceV3UserInputParams(d *schema.ResourceData) map[string]interface{} {
-	return map[string]interface{}{
-		"terraform_resource_data": d,
-		"availability_zone":       d.Get("availability_zone"),
-		"backup_strategy":         d.Get("backup_strategy"),
-		"db":                      d.Get("db"),
-		"flavor":                  d.Get("flavor"),
-		"ha_replication_mode":     d.Get("ha_replication_mode"),
-		"name":                    d.Get("name"),
-		"param_group_id":          d.Get("param_group_id"),
-		"security_group_id":       d.Get("security_group_id"),
-		"subnet_id":               d.Get("subnet_id"),
-		"volume":                  d.Get("volume"),
-		"vpc_id":                  d.Get("vpc_id"),
-		"time_zone":               d.Get("time_zone"),
-		"data_vip":                d.Get("fixed_ip"),
-	}
-}
-
 func resourceRdsInstanceV3Create(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	client, err := config.RdsV3Client(GetRegion(d, config))
+	region := GetRegion(d, config)
+	client, err := config.RdsV3Client(region)
 	if err != nil {
 		return fmt.Errorf("Error creating sdk client, err=%s", err)
 	}
 
-	opts := resourceRdsInstanceV3UserInputParams(d)
-	opts["region"] = GetRegion(d, config)
-	opts["enterprise_project_id"] = GetEnterpriseProjectID(d, config)
+	createRdsOpts := instances.CreateRdsOpts{
+		Name:                d.Get("name").(string),
+		FlavorRef:           d.Get("flavor").(string),
+		VpcId:               d.Get("vpc_id").(string),
+		SubnetId:            d.Get("subnet_id").(string),
+		SecurityGroupId:     d.Get("security_group_id").(string),
+		ConfigurationId:     d.Get("param_group_id").(string),
+		TimeZone:            d.Get("time_zone").(string),
+		FixedIp:             d.Get("fixed_ip").(string),
+		EnterpriseProjectId: GetEnterpriseProjectID(d, config),
+		Region:              region,
+	}
 
-	arrayIndex := map[string]int{
-		"backup_strategy": 0,
-		"db":              0,
-		"volume":          0,
+	// var azList []string
+	azList := make([]string, len(d.Get("availability_zone").([]interface{})))
+	for i, az := range d.Get("availability_zone").([]interface{}) {
+		azList[i] = az.(string)
 	}
+	createRdsOpts.AvailabilityZone = strings.Join(azList, ",")
 
-	params, err := buildRdsInstanceV3CreateParameters(opts, arrayIndex)
-	if err != nil {
-		return fmt.Errorf("Error building the request body of api(create), err=%s", err)
+	if hasFilledOpt(d, "db") {
```

Comment:
as `db` is a required argument, hasFilledOpt will always be true.

Reply:
hasFilledOpt is removed.